### PR TITLE
Suggestion for "change(rpc): Modify `getblocksubsidy` for NU6"

### PR DESCRIPTION
**Pull request by arya2**
This is a suggestion PR for #8742 which includes:
- some optional cleanups/refactors
- removal of the NU6 activation height on Testnet

### CODE REVIEW

File: zebra-rpc/src/methods/tests/snapshot/snapshots/get_block_subsidy_future_nu6_height@testnet_10.snap [hunk 1]
```diff
@@ -1,28 +0,0 @@
...
-      "value": 0.1875,
-      "valueZat": 18750000
-    }
-  ],
-  "miner": 1.25,
-  "founders": 0.0,
-  "fundingstreamstotal": 0.125,
-  "lockboxtotal": 0.1875,
-  "totalblocksubsidy": 1.5625
-}
```
Comment (oxarbitrage) [inline zebra-rpc/src/methods/tests/snapshot/snapshots/get_block_subsidy_future_nu6_height@testnet_10.snap]: Is it possible to run the tests to get this output ?

Comment (arya2) [reply zebra-rpc/src/methods/tests/snapshot/snapshots/get_block_subsidy_future_nu6_height@testnet_10.snap:28]: Yep. I made a mistake in the refactor.

Comment (arya2) [reply zebra-rpc/src/methods/tests/snapshot/snapshots/get_block_subsidy_future_nu6_height@testnet_10.snap]: Yep, I chose a NU6 activation height below of the funding stream height range start bound. Fixed in https://github.com/ZcashFoundation/zebra/pull/8766/commits/1232ff0f1abab90f23b7d42ade19cc2f88228979.

Update: Fixed the block subsidy in the snapshot in https://github.com/ZcashFoundation/zebra/pull/8766/commits/f586bd16ed167d2417feef9c270ba4d9bc3b742c.

File: zebra-rpc/src/methods/tests/snapshot/snapshots/get_block_subsidy_future_nu6_height@nu6testnet_10.snap [hunk 2]
```diff
@@ -7,22 +7,22 @@ expression: get_block_subsidy
...
       "recipient": "Lockbox NU6",
       "specification": "https://zips.z.cash/draft-nuttycom-funding-allocation#alternative-2-hybrid-deferred-dev-fund-transitioning-to-a-non-direct-funding-model",
-      "value": 0.1875,
-      "valueZat": 18750000
+      "value": 0.09375,
+      "valueZat": 9375000
     }
   ],
-  "miner": 1.25,
+  "miner": 0.625,
```
Comment (arya2) [inline zebra-rpc/src/methods/tests/snapshot/snapshots/get_block_subsidy_future_nu6_height@nu6testnet_10.snap]: Ah, I think we need to set the right blossom activation height too.

File: zebra-rpc/src/methods/tests/snapshot.rs [hunk 3]
```diff
@@ -40,10 +41,18 @@ pub const EXCESSIVE_BLOCK_HEIGHT: u32 = MAX_ON_DISK_HEIGHT.0 + 1;
 async fn test_rpc_response_data() {
     let _init_guard = zebra_test::init();
     let default_testnet = Network::new_default_testnet();
+    let nu6_testnet = testnet::Parameters::build()
+        .with_network_name("NU6Testnet")
+        .with_activation_heights(ConfiguredActivationHeights {
+            nu6: Some(POST_NU6_FUNDING_STREAMS_TESTNET.height_range().start.0),
```
Comment (arya2) [inline zebra-rpc/src/methods/tests/snapshot.rs:48]:
```suggestion
            blossom: Some(584_000),
            nu6: Some(POST_NU6_FUNDING_STREAMS_TESTNET.height_range().start.0),
```

File: zebra-rpc/src/methods/tests/snapshot/snapshots/get_difficulty_invalid_populated@mainnet_10.snap
Comment (arya2) [inline zebra-rpc/src/methods/tests/snapshot/snapshots/get_difficulty_invalid_populated@mainnet_10.snap:1]: I think this was an unreferenced snapshot from before this change

File: zebra-rpc/src/methods/tests/snapshots/get_blockchain_info@testnet_10.snap [hunk 4]
```diff
@@ -37,11 +37,6 @@ expression: info
       "name": "NU5",
       "activationheight": 1842420,
       "status": "pending"
-    },
-    "c8e71055": {
-      "name": "Nu6",
-      "activationheight": 2942000,
-      "status": "pending"
```
Comment (oxarbitrage) [inline zebra-rpc/src/methods/tests/snapshots/get_blockchain_info@testnet_10.snap:44]: Do we want to do the same as the getblocksubsidy and have a snapshot with this included ?

Comment (arya2) [reply zebra-rpc/src/methods/tests/snapshots/get_blockchain_info@testnet_10.snap:44]: Yeah, we'll need it shortly, added in https://github.com/ZcashFoundation/zebra/pull/8766/commits/4dba1e8746cbcd93e4f660927052487b68347184.

**Review by oxarbitrage — APPROVED** [review]
Thanks!

**Pull request merged**